framework/Source/PictureInput.swift

```diff
@@ -54,41 +58,77 @@ public class PictureInput: ImageSource {
                     internalImage = nil
-                    self.internalTexture = Texture(
-                        orientation: .portrait,
-                        texture: imageTexture,
-                        timingStyle: isTransient ? .transientImage : .stillImage
-                    )
-                    self.updateTargetsWithTexture(self.internalTexture!)
+                    let texture = makeTexture(with: mtlTexture)
+                    self.internalTexture = texture
+                    self.updateTargetsWithTexture(texture)
                     self.hasProcessedImage = true
                 } catch {
                     fatalError("Failed loading image texture")
                 }
             } else {
-                textureLoader.newTexture(cgImage: internalImage!, options: [MTKTextureLoader.Option.SRGB : false], completionHandler: { [weak self] (possibleTexture, error) in
+                textureLoader.newTexture(cgImage: internalImage!, options: newTextureOptions, completionHandler: { [weak self] (possibleTexture, error) in
                     guard let self else { return }
                     guard (error == nil) else { fatalError("Error in loading texture: \(error!)") }
-                    guard let texture = possibleTexture else { fatalError("Nil texture received") }
+                    guard let mtlTexture = possibleTexture else { fatalError("Nil texture received") }
                     self.internalImage = nil
-                    self.internalTexture = Texture(
-                        orientation: .portrait,
-                        texture: texture,
-                        timingStyle: self.isTransient ? .transientImage : .stillImage
-                    )
+
+                    let texture = makeTexture(with: mtlTexture)
+                    self.internalTexture = texture
                     DispatchQueue.global().async{
-                        self.updateTargetsWithTexture(self.internalTexture!)
+                        self.updateTargetsWithTexture(texture)
                         self.hasProcessedImage = true
                     }
                 })
             }
         }
     }
-    
+
+    @available(iOS 13, *)
+    public func processImage() async {
+        if let texture = internalTexture {
+            updateTargetsWithTexture(texture)
+            hasProcessedImage = true
+            return
+        }
+
+        guard let internalImage else { return }
+        let textureLoader = MTKTextureLoader(device: sharedMetalRenderingDevice.device)
+        do {
+            let mtlTexture = try await textureLoader.newTexture(
+                cgImage:internalImage,
+                options: newTextureOptions
+            )
+            self.internalImage = nil
+
+            let texture = makeTexture(with: mtlTexture)
+            internalTexture = texture
+            updateTargetsWithTexture(texture)
```

Comment:
Should we set `hasProcessedImage = true`?

Reply:
Yes we should; thanks!!
I'll correct this in the next commit!